expression/testdata/partition_pruner_out.json

```diff
@@ -382,26 +382,26 @@
         "SQL": "select * from t1 where a=id and id >10",
         "Result": null,
         "Plan": [
-          "TableReader_7 2666.67 root partition:all data:Selection_6",
-          "└─Selection_6 2666.67 cop[tikv]  eq(test_partition.t1.a, test_partition.t1.id), gt(test_partition.t1.id, 10)",
+          "TableReader_7 888.89 root partition:all data:Selection_6",
+          "└─Selection_6 888.89 cop[tikv]  eq(test_partition.t1.a, test_partition.t1.id), gt(test_partition.t1.a, 10), gt(test_partition.t1.id, 10)",
```

Comment:
We only need two of three expressions from `eq(test_partition.t1.a, test_partition.t1.id), gt(test_partition.t1.a, 10), gt(test_partition.t1.id, 10)` to determine the result.

Reply:
I have discussed it with @ou-bing . The reason is `PropagateConstant` needs to keep the origin equal condition for some other optimization like join operators. We can improve it in another PR.

Reply:
@winoros What's your opinion? PTAL.

Reply:
For this three expression, the `t1.a=t1.id` must be kept and at least one of the `t1.a > 10` and `t1.id > 10` should be kept.
If we remove `t1.a=t1.id` and keep the other two, the result will be obviously wrong.

We can keep its behavior here and do the optimization later. This should be a improvement for the `ConstantPropagation` itself.